flycheck.el

```diff
@@ -5264,7 +5264,16 @@ function `buffer-file-name'."
       (when (seq-some (apply-partially #'flycheck-same-files-p
                                        (expand-file-name filename cwd))
                       buffer-files)
-        (setf (flycheck-error-filename err) (buffer-file-name)))))
+        (let ((new-filename (buffer-file-name)))
```

Comment:
Are we sure that (buffer-file-name) is non-nil here? Otherwise the `replace-regexp-in-string` part will error out.